File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/fragments/LoginScanFragment.java
package de.dimedis.mobileentry.fragments;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

import de.dimedis.mobileentry.backend.BackendService;
import de.dimedis.mobileentry.backend.response.UserLoginByBarcodeResponse;
import de.dimedis.mobileentry.bbapi.BarcodeManager;
import de.dimedis.mobileentry.databinding.FragmentLoginScanBinding;
import de.dimedis.mobileentry.util.CommonUtils;
import de.dimedis.mobileentry.util.SessionUtils;

public class LoginScanFragment extends LoginFragment {

    private FragmentLoginScanBinding binding;

    public static Builder builder() {
        return new Builder();
    }

    public static class Builder {
        public Builder() {
        }

        public LoginScanFragment build() {
            return new LoginScanFragment();
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        binding = FragmentLoginScanBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
    }

    @Override
    protected TextView getErrorMessage() {
        return binding.errorMessage;
    }

    @Override
    public void onStart() {
        super.onStart();
        BarcodeManager.getInstance().open(this.getActivity());
    }

    @Override
    public void onStop() {
        super.onStop();
        BarcodeManager.getInstance().close(this.getActivity());
    }

    private void loginByBarcode(String barcode) {
        if (TextUtils.isEmpty(barcode)) {
            showError(true);
        } else {
            if (!CommonUtils.isInternetConnected()) {
                SessionUtils.loginOffline(barcode);
            } else {
                BackendService.userLoginByBarcode(barcode);
                setVisibilityProgressBar(true);
            }
        }
    }

    @Override
    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onEventMainThread(UserLoginByBarcodeResponse event) {
        super.onEventMainThread(event);
        restartScreenSaverTask();
    }

    @SuppressWarnings("unused")
    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onEventMainThread(BarcodeManager.BarcodeScannedEvent event) {
        loginByBarcode(event.barcode);
        restartScreenSaverTask();
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/fragments/LoginScanFragment.kt
package de.dimedis.mobileentry.fragments

import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import de.dimedis.mobileentry.backend.BackendServiceUtil
import de.dimedis.mobileentry.backend.response.UserLoginByBarcodeResponse
import de.dimedis.mobileentry.bbapi.BarcodeManager
import de.dimedis.mobileentry.bbapi.BarcodeManager.BarcodeScannedEvent
import de.dimedis.mobileentry.databinding.FragmentLoginScanBinding
import de.dimedis.mobileentry.util.CommonUtils.isInternetConnected
import de.dimedis.mobileentry.util.SessionUtils.loginOffline
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

class LoginScanFragment : LoginFragment() {
    private var binding: FragmentLoginScanBinding? = null

    class Builder {
        fun build(): LoginScanFragment {
            return LoginScanFragment()
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        super.onCreateView(inflater, container, savedInstanceState)
        binding = FragmentLoginScanBinding.inflate(inflater, container, false)
        return binding!!.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
    }

    override fun getErrorMessage(): TextView {
        return binding!!.errorMessage
    }

    override fun onStart() {
        super.onStart()
        this.activity?.let { BarcodeManager.instance.open(it) }
    }

    override fun onStop() {
        super.onStop()
        this.activity?.let { BarcodeManager.instance.close(it) }
    }

    private fun loginByBarcode(barcode: String) {
        if (TextUtils.isEmpty(barcode)) {
            showError(true)
        } else {
            if (!isInternetConnected()) {
                loginOffline(barcode)
            } else {
                BackendServiceUtil.userLoginByBarcode(barcode)
                setVisibilityProgressBar(true)
            }
        }
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    override fun onEventMainThread(event: UserLoginByBarcodeResponse) {
        super.onEventMainThread(event)
        restartScreenSaverTask()
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    fun onEventMainThread(event: BarcodeScannedEvent) {
        event.barcode?.let { loginByBarcode(it) }
        restartScreenSaverTask()
    }

    companion object {
        fun builder(): Builder {
            return Builder()
        }
    }
}